node/src/actors/chain_manager/mod.rs

```diff
@@ -278,67 +281,88 @@ impl ChainManager {
-                        transaction: reveal,
-                    })
-                }
-                // Persist finished data requests into storage
-                let to_be_stored = self.chain_state.data_request_pool.finished_data_requests();
-                to_be_stored.into_iter().for_each(|dr| {
-                    self.persist_data_request(ctx, &dr);
-                    if info_flag {
+                if let StateMachine::Synced = self.sm_state {
+                    // Persist finished data requests into storage
+                    let to_be_stored = self.chain_state.data_request_pool.finished_data_requests();
+                    to_be_stored.into_iter().for_each(|dr| {
+                        self.persist_data_request(ctx, &dr);
                         show_info_tally(&self.chain_state.unspent_outputs_pool, dr, block_epoch);
-                    }
-                });
+                    });
 
-                if info_flag {
                     show_info_dr(&self.chain_state.data_request_pool, &block);
 
-                    debug!("{:?}", block);
+                    log::trace!("{:?}", block);
                     debug!("Mint transaction hash: {:?}", block.txns[0].hash());
-                }
 
-                // Insert candidate block into `block_chain` and persist it
-                self.chain_state.block_chain.insert(block_epoch, block_hash);
-                self.persist_item(ctx, InventoryItem::Block(block.clone()));
+                    let reveals = self
+                        .chain_state
+                        .data_request_pool
+                        .update_data_request_stages();
+
+                    for reveal in reveals {
+                        // Send AddTransaction message to self
+                        // And broadcast it to all of peers
+                        ctx.address().do_send(AddTransaction {
+                            transaction: reveal,
+                        })
+                    }
+                    self.persist_item(ctx, InventoryItem::Block(block.clone()));
 
-                // Persist chain_info into storage
-                if let StateMachine::Synced = self.sm_state {
+                    // Persist chain_info into storage
                     self.persist_chain_state(ctx);
-                }
 
-                // Send notification to JsonRpcServer
-                JsonRpcServer::from_registry().do_send(NewBlock { block })
+                    // Send notification to JsonRpcServer
+                    JsonRpcServer::from_registry().do_send(NewBlock {
+                        block: block.clone(),
+                    })
+                }
             }
             None => {
                 error!("No ChainInfo loaded in ChainManager");
             }
         }
     }
+
+    fn get_chain_beacon(&self) -> CheckpointBeacon {
+        self.chain_state
+            .chain_info
+            .as_ref()
+            .unwrap()
```

Comment:
Can we avoid the `unwrap` and rather handle the `None` case where used?

Reply:
We could and I did previously, but I reverted the change because I didn't want to introduce too many changes. Problem is that the invariant for the actor's state and the state machine states is not that clear and there are a lot of unnecessary checks for optional values that inside the synchronization/candidate logic aren't optional anymore.

Instead of just changing this unwrap I would prefer to leave it as it is and refactor this actor in the future to avoid the unnecessary checks, e.g.: [let Some(...)](https://github.com/witnet/witnet-rust/pull/577/files/95916ff11b761ef16975ac9f9f31786a01f13414#diff-089d6d21f03cb529865367bff4779506R192), [match self.chain_state.chain_info.as_mut()](https://github.com/witnet/witnet-rust/blob/95916ff11b761ef16975ac9f9f31786a01f13414/node/src/actors/chain_manager/mod.rs#L274) and so on